components/fxa-client/src/fxa_client.udl

```diff
@@ -723,6 +709,29 @@ interface FirefoxAccount {
   
 };
 
+dictionary FxaConfig {
+    // FxaServer to connect with
+    FxaServer server;
+    // Registered OAuth client id of the application.
+    string client_id;
+    // `redirect_uri` - the registered OAuth redirect URI of the application.
+    string redirect_uri;
+    //  URL for the user's Sync Tokenserver. This can be used to support users who self-host their
+    //  sync data. If `None` then it will default to the Mozilla-hosted Sync server.
+    string? token_server_url_override;
+};
+
+// FxA server to connect to
+[Enum]
+interface FxaServer {
+    Release();
+    Stable();
+    Stage();
+    China();
+    LocalDev();
+    Custom(string content_url);
```

Comment:
How do you feel about using `Url` here? As a nit, I'd say just `url` is better than `content_url`, particularly for external people who might misunderstand what `content` means in this context.

Reply:
`url` makes more sense to me than `content_url`.  I'll make that change.

I tested out the using `Url` directly and the main issue is that parsing URLs is failing.  I think that means we need to either make the constructor failing  or panic in the face of invalid URLs.  I'm thinking we can hold off on this one for now.